### PR TITLE
feat(0000): questions improved

### PR DESCRIPTION
**Pull request by alexZogich**
## PR Checklist

PR meets the following requirements:

- [x] Changes follow our [guidelines](https://github.com/extreme-startup/questionator/wiki/Contributing)
- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

## PR Type

PR introduces the following changes:

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## Actual/New behavior

Issue Number: N/A

### CODE REVIEW

File: server/src/modules/Question/dto/question.dto.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export class QuestionDto {
```
Comment (kovnick) [inline server/src/modules/Question/dto/question.dto.ts]: QuestionDto moved to Question/question.dto.ts. This is the old one.
BTW do we need contestId inside dto object?

File: server/src/modules/Question/question.service.ts [hunk 2]
```diff
@@ -39,26 +39,18 @@ export class QuestionService {
         }
     }
 
-    async insert(question: QuestionDto): Promise<QuestionDto> {
+    async insert(question: QuestionDto): Promise<Question> {
```
Comment (kovnick) [inline server/src/modules/Question/question.service.ts]: I think here should be QuestionDto.

File: server/src/modules/Question/question.service.ts [hunk 3]
```diff
@@ -39,26 +39,18 @@ export class QuestionService {
...
         const newQuestion = new Question();
```
Comment (kovnick) [inline server/src/modules/Question/question.service.ts]: const newQuestion = this.questionRepository.create(question);     
.....     
return this.questionRepository.save(newQuestion);       

https://github.com/typeorm/typeorm/blob/master/docs/repository-api.md

File: server/src/modules/Question/question.service.ts [hunk 4]
```diff
@@ -39,26 +39,18 @@ export class QuestionService {
...
         }
     }
 
     async update(oldQuestion: Question, updatedValues: QuestionDto): Promise<Question> {
-        const updatedQuestion = oldQuestion;
-
-        Object.keys(updatedValues).forEach((key) => {
-            updatedQuestion[key] = updatedValues[key];
-        });
+        const updatedQuestion = { ...oldQuestion, ...updatedValues };
```
Comment (kovnick) [inline server/src/modules/Question/question.service.ts]: same here

File: server/src/migration/1553647346431-QuestionMigration.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ export class Question1553647346431 implements MigrationInterface {
     async up(queryRunner: QueryRunner): Promise<any> {
         await queryRunner.createTable(
           new Table({
-            name: 'qms_question',
+            name: 'question',
```
[the remark this answers is not in the document]
Comment (alexZogich) [reply server/src/migration/1553647346431-QuestionMigration.ts]: removed that

File: server/src/migration/1553936929517-SetQuestionIsDeletedDefaultValue.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { MigrationInterface, QueryRunner } from 'typeorm';
 export class SetQuestionIsDeletedDefaultValue1553936929517 implements MigrationInterface {
 
     public async up(queryRunner: QueryRunner): Promise<any> {
-        const table = await queryRunner.getTable('qms_question');
+        const table = await queryRunner.getTable('question');
```
[the remark this answers is not in the document]
Comment (alexZogich) [reply server/src/migration/1553936929517-SetQuestionIsDeletedDefaultValue.ts]: removed that

File: server/src/migration/1554464474584-Question.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import {MigrationInterface, QueryRunner} from 'typeorm';
+
+export class Question1554464474584 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        await queryRunner.query('ALTER TABLE `question` ADD `contestId` int NULL');
+        await queryRunner.query('ALTER TABLE `question` CHANGE `isDeleted` `isDeleted` tinyint NOT NULL DEFAULT 0');
+        await queryRunner.query('ALTER TABLE `contest` CHANGE `isDeleted` `isDeleted` tinyint NOT NULL');
+        await queryRunner.query('ALTER TABLE `customer` DROP PRIMARY KEY');
```
[the remark this answers is not in the document]
Comment (alexZogich) [reply server/src/migration/1554464474584-Question.ts]: They are related and autogenerated by typeorm

File: server/src/modules/Question/dto/question.dto.ts (hunk 1 above)
Comment (alexZogich) [reply server/src/modules/Question/dto/question.dto.ts]: removed

File: server/src/modules/Question/question.service.ts (hunk 2 above)
Comment (alexZogich) [reply server/src/modules/Question/question.service.ts]: changed for consistency purposes for whole file

File: server/src/modules/Question/question.service.ts (hunk 3 above)
Comment (alexZogich) [reply server/src/modules/Question/question.service.ts]: fixed

File: server/src/modules/Question/question.service.ts (hunk 4 above)
Comment (alexZogich) [reply server/src/modules/Question/question.service.ts]: fixed

**Pull request merged**